PluginWindow.cs

```diff
@@ -286,6 +287,31 @@ public void Draw(out bool manageGroups, out bool close)
             if (ImGui.IsItemHovered(ImGuiHoveredFlags.AllowWhenDisabled) && !ImGui.GetIO().KeyCtrl)
                 ImGui.SetTooltip("Hold CTRL to discard any changes you've made since you've last saved.");
 
+            ImGui.SameLine();
+            bool randomAssign = Service.Configuration.assignRandomTitleOnAreaChange;
+            if (ImGui.Checkbox("Random title when switching Zones", ref randomAssign))
+                save = true;
+            Service.Configuration.assignRandomTitleOnAreaChange = randomAssign;
```

Comment:
both the save and the config update should only happen as part of the `if` block